Heimdallr/iOSHeimdallr.swift

```diff
@@ -0,0 +1,131 @@
+import Foundation
+import Result
+import UIKit
+
+public extension Heimdallr {
+
+    /// Requests an access token with the authorization code grant.
+    ///
+    /// **Note:** The completion closure may be invoked on any thread.
+    ///
+    /// - Parameters:
+    ///   - url: The authorization code link.
+    ///   - redirectURI: The redirect URI.
+    ///   - scope: The authorization scope.
+    ///   - completion: A cllback to invoke when the request completed.
+    public func requestAccessToken(authorizationCodeURL url: URL,
+                                   redirectURI: String,
+                                   scope: String,
+                                   parameters: [String: String],
+                                   completion: @escaping (Result<OAuthAccessToken, NSError>) -> Void) {
+
+        requestAuthorizationCode(authorizationCodeURL: url,
+                                 redirectURI: redirectURI,
+                                 scope: scope,
+                                 parameters: parameters) { [weak self] result in
+
+            guard let strongSelf = self else {
+                completion(.failure(NSError(domain: "", code: 0, userInfo: nil)))
+                return
+            }
+
+            switch result {
+            case let .success(code):
+                strongSelf.requestAccessToken(authorizationCode: code, redirectURI: redirectURI, completion: completion)
+            case let .failure(error):
+                completion(.failure(error))
+            }
+        }
+    }
+
+    /// Requests an access token with the Implicit grant.
+    ///
+    /// - Parameters:
+    ///   - authorizationCodeURL: The authorization code link.
+    ///   - redirectURI: The redirect URI.
+    ///   - scope: The authorization scope.
+    ///   - completion: A cllback to invoke when the request completed.
+    public func requestAccessToken(implicitAuthorizationURL url: URL,
+                                   redirectURI: String,
+                                   scope: String,
+                                   completion: @escaping (Result<OAuthAccessToken, NSError>) -> Void) {
+        requestAccessToken(implicitAuthorizationURL: url,
+                           redirectURI: redirectURI,
+                           scope: scope,
+                           responseType: "token",
+                           completion: completion)
+    }
+
+    /// Requests an authorization code.
+    ///
+    /// - Parameters:
+    ///   - authorizationCodeURL: The authorization code link.
+    ///   - redirectURI: The redirect URI.
+    ///   - scope: The authorization scope.
+    ///   - completion: A cllback to invoke when the request completed.
+    public func requestAuthorizationCode(authorizationCodeURL url: URL,
```

Comment:
Maybe rename to sth. like `requestAuthorizationTypeCode`

Reply:
I think `requestAuthorizationCode ` is allgined to `requestAccessToken`.